data/updater.py

```diff
@@ -0,0 +1,357 @@
+        raise UpdateFailed
+
+    current_version_dir = os.path.dirname(CURRENT_VERSION_PATH)
+    try:
+        os.makedirs(current_version_dir)
+    except OSError as ose:
+        if ose.errno != errno.EEXIST:
+            raise
+
+    # Update the current version number
+    with open(CURRENT_VERSION_PATH, "w+") as out_file:
+        out_file.write(str(version))
+
+    logging.info("Updated to version {0}".format(version))
+
+def update_to_version(from_version, to_version, skip_verification=False):
+    versions = range(from_version + 1, to_version + 1)
+    for version in versions:
+        try:
+            perform_update(version, skip_verification)
+        except UpdateFailed:
+            logging.error("Failed to update to version {0}".format(version))
+            return
+
+def check_for_update(skip_verification=False):
+    logging.info("Checking for update")
+    current_version = get_current_version()
+    try:
+        latest_version = get_latest_version()
+    except RequestFailed:
+        logging.error("Failed to learn the latest version")
+        return
+
+    if current_version < latest_version:
+        logging.info("Updating {0}->{1}".format(current_version, latest_version))
+        update_to_version(current_version, latest_version, skip_verification)
+    else:
+        logging.info("Already up to date")
+
+class InvalidInterval(Exception):
+    pass
+
+def _get_interval(interval):
+    """
+    Returns the interval in seconds.
+    """
+    seconds = 0
+    INTERVAL_REGEXP = re.compile("(\d+d)?(\d+h)?(\d+m)?")
+    m = INTERVAL_REGEXP.match(interval)
+    days, hours, minutes = m.groups()
+
+    if days is not None:
+        seconds += int(days[:-1]) * 24 * 60 * 60
+    if hours is not None:
+        seconds += int(hours[:-1]) * 60 * 60
+    if minutes is not None:
+        seconds += int(minutes[:-1]) * 60
+
+    if seconds == 0:
+        try:
+            seconds = int(interval)
+        except ValueError:
+            raise InvalidInterval
+    return seconds
+
+
+def update(args):
+    """
+    This command fires the updater.
+    """
+    if args.watch is True:
+        seconds = _get_interval(args.interval)
+        while True:
+            check_for_update(skip_verification=args.skip_verification)
+            time.sleep(seconds)
+    else:
+        check_for_update(skip_verification=args.skip_verification)
+
+
+def install(args):
+    """
+    This command installs the updater.
+    """
+    directories = [
+        UPDATER_PATH,
+        os.path.dirname(CURRENT_VERSION_PATH)
+    ]
+    for path in directories:
+        try:
+            os.makedirs(path)
+        except OSError as ose:
+            if ose.errno != errno.EEXIST:
+                raise
+
+    with open(CURRENT_VERSION_PATH, "w") as out_file:
+        out_file.write("0")
+
+    # Copy myself over to the SCRIPT_INSTALL_PATH
+    shutil.copyfile(__file__, SCRIPT_INSTALL_PATH)
+    os.chmod(SCRIPT_INSTALL_PATH, int('744', 8))
```

Comment:
That is mainly to make it easier to understand the permissions you set, since usually you reason in terms of base8 permissions (like when you run chmod from the command line), while os.chmod takes a integer. It's more readable to see `os.chmod(xx, int('744', 8))` than `os.chmod(xx, 484)`